packages/legend-query-builder/src/components/QueryLoader.tsx

```diff
@@ -0,0 +1,592 @@
+              <button
+                className={clsx('query-loader__filter__toggler__btn', {
+                  'query-loader__filter__toggler__btn--toggled': isMineOnly,
+                })}
+                onClick={toggleShowCurrentUserQueriesOnly}
+                tabIndex={-1}
+              >
+                Mine Only
+              </button>
+              {queryLoaderState.extraFilterOptions.length > 0 && (
+                <div className="query-loader__filter__extra__filters">
+                  {Array.from(queryLoaderState.extraFilters.entries()).map(
+                    ([key, value]) => (
+                      <button
+                        key={key}
+                        className={clsx('query-loader__filter__toggler__btn', {
+                          'query-loader__filter__toggler__btn--toggled': value,
+                        })}
+                        onClick={(): void => toggleExtraFilters(key)}
+                        tabIndex={-1}
+                      >
+                        {key}
+                      </button>
+                    ),
+                  )}
+                </div>
+              )}
+            </div>
+          </div>
+        </div>
+        <div className="query-loader__content">
+          <PanelLoadingIndicator
+            isLoading={
+              !queryLoaderState.searchQueriesState.hasCompleted ||
+              renameQueryState.isInProgress
+            }
+          />
+
+          <div className="query-loader__results">
+            {queryLoaderState.searchQueriesState.hasCompleted && (
+              <>
+                <div className="query-loader__results__summary">
+                  {queryLoaderState.showingRecentlyViewedQueries ? (
+                    results.length ? (
+                      `Showing ${quantifyList(
+                        results,
+                        'recently viewed query',
+                        'recently viewed queries',
+                      )}`
+                    ) : (
+                      `No recently viewed queries`
+                    )
+                  ) : results.length >= QUERY_LAODER_TYPEAHEAD_SEARCH_LIMIT ? (
+                    <>
+                      {`Found ${QUERY_LAODER_TYPEAHEAD_SEARCH_LIMIT}+ matches`}{' '}
+                      <InfoCircleIcon
+                        title="Some queries are not listed, refine your search to get better matches"
+                        className="query-loader__results__summary__info"
+                      />
+                    </>
+                  ) : (
+                    `Found ${quantifyList(results, 'match', 'matches')}`
+                  )}
+                </div>
+                {results
+                  .slice(0, QUERY_LAODER_TYPEAHEAD_SEARCH_LIMIT)
+                  .map((query, idx) => (
+                    <div
+                      className="query-loader__result"
+                      title={
+                        showQueryNameEditInput !== idx
```

Comment:
this kind of scenario, when you have nested buttons, it's better to use `stopPropagation` rather than doing the `idx !== showQueryNameEditInput` like this.